java/arcs/core/storage/StorageProxy.kt

```diff
@@ -37,57 +37,50 @@ class StorageProxy<Data : CrdtData, Op : CrdtOperation, T>(
     initialCrdt: CrdtModel<Data, Op, T>
 ) {
     private val log = TaggedLog { "StorageProxy" }
-    private val mutex = reentrantLock()
+    private val handlesMutex = Mutex()
     private val handles: MutableSet<Handle<Data, Op, T>> = mutableSetOf()
+    private var hasReaders = false
+    private val syncMutex = Mutex()
     private val crdt = initialCrdt
```

Comment:
Should we use guard again now that we're back on the normal mutex path?

Reply:
Yup, thanks!